### ISSUE TITLE
Server crash on error in the `PRUNE` expression

### ISSUE THREAD

**Issue by wajda** [arangodb/arangodb#17291]
## My Environment

* __ArangoDB Version__: 3.9.3
* __Deployment Mode__:  Single
* __Infrastructure__:     localhost
* __Operating System__:     Fedora Linux 36
* __Total RAM in your machine__:     32Gb
* __Disks in use__:    SSD
* __Used Package__:    rpm

## Component, Query & Data

__Affected feature__: AQL, Graph traversal with `PRUNE` on a user defined function  -->

__AQL query (if applicable)__:
```aql
WITH Foo, Bar
FOR v IN 1 OUTBOUND {"_id": "Foo/1"} Bar
    PRUNE SPLINE::sdlkfjglskdjfhglkdsjfhglkjsdfh()
    RETURN v
```
__AQL explain and/or profile (if applicable)__:
```
Query String (118 chars, cacheable: false):
 WITH Foo, Bar
 FOR v IN 1 OUTBOUND {"_id": "Foo/1"} Bar
     PRUNE SPLINE::sdlkfjglskdjfhglkdsjfhglkjsdfh()
     RETURN v

Execution plan:
 Id   NodeType        Est.   Comment
  1   SingletonNode      1   * ROOT
  2   TraversalNode      1     - FOR v  /* vertex */ IN 1..1  /* min..maxPathDepth */ OUTBOUND 'Foo/1' /* startnode */  Bar
  3   ReturnNode         1       - RETURN v

Indexes used:
 By   Name   Type   Collection   Unique   Sparse   Selectivity   Fields        Ranges
  2   edge   edge   Bar          false    false       100.00 %   [ `_from` ]   base OUTBOUND

Traversals on graphs:
 Id  Depth  Vertex collections  Edge collections  Options                                  Filter / Prune Conditions                                                   
 2   1..1                       Bar               uniqueVertices: none, uniqueEdges: path  PRUNE SPLINE::SDLKFJGLSKDJFHGLKDSJFHGLKJSDFH()   /* user-defined function */

Optimization rules applied:
 none

Optimization rules with highest execution times:
 RuleName                                    Duration [s]
 optimize-traversals                              0.00001
 use-indexes                                      0.00000
 splice-subqueries                                0.00000
 replace-function-with-index                      0.00000
 optimize-count                                   0.00000

40 rule(s) executed, 1 plan(s) created

```

__Dataset__: empty

## Steps to reproduce

1. Create an empty vertex collection `Foo` and an empty edge collection `Bar`
2. Execute a query above (the function `SPLINE::SDLKFJGLSKDJFHGLKDSJFHGLKJSDFH()` does not exist)

__Problem__:
The whole server goes down with an error below.
The problem happens where there is a runtime error in a `PRUNE` clause. It doesn't matter if a user function throws an error, or you simply call a non-existing user function.

```
2022-10-06T08:41:14Z [87039] FATAL [a7902] {crash} 💥 ArangoDB 3.9.3 [linux], thread 4 [SchedWorker] caught unexpected signal 11 (SIGSEGV) accessing address 0x0000000000000008: signal handler invoked - image base address: 0x0000000000400000 - CPU context: rip: 0x0000000000f825f0, rsp: 0x00007f39e8ffca20, efl: 0x0000000000010202, rbp: 0x00007f39e8ffcb70, rsi: 0x00007f39ca00a4d8, rdi: 0x0000000000000003, rax: 0x0000000000000000, rbx: 0x00007f39ca0c8180, rcx: 0x0000000000000030, rdx: 0x00007f39e8ffcbcf, r8: 0x00007f39a27fc328, r9: 0x00007f39ca007300, r10: 0x00007f39e8ffcba0, r11: 0x000000000000001b, r12: 0x00007f39e8ffcc00, r13: 0x00007f39ca00a4d8, r14: 0x0000000000000000, r15: 0x00000000000003e8
2022-10-06T08:41:14Z [87039] INFO [c962b] {crash} Backtrace of thread 4 [SchedWorker]
2022-10-06T08:41:14Z [87039] INFO [308c3] {crash} frame  1 [+0x00000000016bd8c5] (anonymous namespace)::crashHandlerSignalHandler(int, siginfo_t*, void*) (+0x35)
2022-10-06T08:41:14Z [87039] INFO [308c3] {crash} frame  2 [+0x00000000039ac5a5] sigprocmask (+0x21)
2022-10-06T08:41:14Z [87039] INFO [308c3] {crash} frame  3 [+0x0000000000d909aa] arangodb::traverser::DepthFirstEnumerator::shouldPrune() (+0x3da)
2022-10-06T08:41:14Z [87039] INFO [308c3] {crash} frame  4 [+0x0000000000d91051] arangodb::traverser::DepthFirstEnumerator::next() (+0x221)
2022-10-06T08:41:14Z [87039] INFO [308c3] {crash} frame  5 [+0x0000000000ce9752] arangodb::aql::TraversalExecutor::doOutput(arangodb::aql::OutputAqlItemRow&) (+0x82)
2022-10-06T08:41:14Z [87039] INFO [308c3] {crash} frame  6 [+0x0000000000aec0ec] arangodb::aql::ExecutionBlockImpl<arangodb::aql::TraversalExecutor>::executeWithoutTrace(arangodb::aql::AqlCallStack const&) (+0x20c)
2022-10-06T08:41:14Z [87039] INFO [308c3] {crash} frame  7 [+0x0000000000aeef25] arangodb::aql::ExecutionBlockImpl<arangodb::aql::TraversalExecutor>::execute(arangodb::aql::AqlCallStack const&) (+0xb5)
2022-10-06T08:41:14Z [87039] INFO [308c3] {crash} frame  8 [+0x00000000019820de] arangodb::aql::SingleRowFetcher<(arangodb::aql::BlockPassthrough)1>::execute(arangodb::aql::AqlCallStack&) [clone .isra.0] (+0x90e)
2022-10-06T08:41:14Z [87039] INFO [308c3] {crash} frame  9 [+0x0000000000a6cd87] arangodb::aql::ExecutionBlockImpl<arangodb::aql::IdExecutor<arangodb::aql::SingleRowFetcher<(arangodb::aql::BlockPassthrough)1> > >::executeFetcher(arangodb::aql::ExecutionBlockImpl<arangodb::aql::IdExecutor<arangodb::aql::SingleRowFetcher<(arangodb::aql::BlockPassthrough)1> > >::ExecutionContext&, arangodb::aql::AqlCall const&) (+0x107)
2022-10-06T08:41:14Z [87039] INFO [308c3] {crash} frame 10 [+0x0000000000a6f4a0] arangodb::aql::ExecutionBlockImpl<arangodb::aql::IdExecutor<arangodb::aql::SingleRowFetcher<(arangodb::aql::BlockPassthrough)1> > >::executeWithoutTrace(arangodb::aql::AqlCallStack const&) (+0x1d70)
2022-10-06T08:41:14Z [87039] INFO [308c3] {crash} frame 11 [+0x0000000000a82915] arangodb::aql::ExecutionBlockImpl<arangodb::aql::IdExecutor<arangodb::aql::SingleRowFetcher<(arangodb::aql::BlockPassthrough)1> > >::execute(arangodb::aql::AqlCallStack const&) (+0xb5)
2022-10-06T08:41:14Z [87039] INFO [308c3] {crash} frame 12 [+0x00000000009601e6] arangodb::aql::Query::execute(arangodb::aql::QueryResult&) (+0x7a6)
2022-10-06T08:41:14Z [87039] INFO [308c3] {crash} frame 13 [+0x000000000066d069] arangodb::RestCursorHandler::registerQueryOrCursor(arangodb::velocypack::Slice const&) (+0x1429)
2022-10-06T08:41:14Z [87039] INFO [308c3] {crash} frame 14 [+0x000000000066db5c] arangodb::RestCursorHandler::execute() (+0x32c)
2022-10-06T08:41:14Z [87039] INFO [308c3] {crash} frame 15 [+0x00000000005c1496] arangodb::rest::RestHandler::executeEngine(bool) (+0xb6)
2022-10-06T08:41:14Z [87039] INFO [308c3] {crash} frame 16 [+0x00000000005c54fa] arangodb::rest::RestHandler::runHandlerStateMachine() (+0x3fa)
2022-10-06T08:41:14Z [87039] INFO [308c3] {crash} frame 17 [+0x00000000008cd8de] arangodb::Scheduler::WorkItem<arangodb::rest::CommTask::handleRequestAsync(std::shared_ptr<arangodb::rest::RestHandler>, unsigned long*)::{lambda()#1}>::invoke() (+0xfe)
2022-10-06T08:41:14Z [87039] INFO [308c3] {crash} frame 18 [+0x0000000000724b8f] arangodb::SupervisedScheduler::runWorker() (+0x2df)
2022-10-06T08:41:14Z [87039] INFO [308c3] {crash} frame 19 [+0x00000000016d15bd] arangodb::Thread::startThread(void*) (+0x6d)
2022-10-06T08:41:14Z [87039] INFO [308c3] {crash} frame 20 [+0x000000000172077a] ThreadStarter(void*) [clone .lto_priv.0] (+0xaa)
2022-10-06T08:41:14Z [87039] INFO [308c3] {crash} frame 21 [+0x00000000039b33dc] start (+0x70)
2022-10-06T08:41:14Z [87039] INFO [ded81] {crash} available physical memory: 33258086400, rss usage: 753094656, vsz usage: 2321113088, threads: 54

```

__Expected result__:
The given query is expected to fail with a meaningful error message, but the server itself must not be affected by an error in an individual AQL query.

**Comment by wajda** [comment]
Hm, it also happens when the user function is OK and is called properly.
```
WITH progress, progressOf
FOR v IN 1 OUTBOUND "progress/123123123" progressOf
    PRUNE SPLINE::IS_VISIBLE_FROM_TX({"num": 26, "liveTxIds": []})
    RETURN v
```
In this case there is no error in the `PRUNE` expression, but the server fails anyway

```
2022-10-06T09:51:01Z [96373] FATAL [a7902] {crash} 💥 ArangoDB 3.9.3 [linux], thread 4 [SchedWorker] caught unexpected signal 11 (SIGSEGV) accessing address 0x0000000000000008: signal handler invoked - image base address: 0x0000000000400000 - CPU context: rip: 0x0000000000f825f0, rsp: 0x00007f3dbd9fca20, efl: 0x0000000000010202, rbp: 0x00007f3dbd9fcb70, rsi: 0x00007f3da14b0580, rdi: 0x0000000000000003, rax: 0x0000000000000000, rbx: 0x00007f3d6a625360, rcx: 0x0000000000000030, rdx: 0x00007f3dbd9fcbcf, r8: 0x00007f3d7868e300, r9: 0x00007f3d6b632d00, r10: 0x00007f3dbd9fcba0, r11: 0x0000000000000006, r12: 0x00007f3dbd9fcc00, r13: 0x00007f3da14b0580, r14: 0x0000000000000000, r15: 0x00000000000003e8
2022-10-06T09:51:01Z [96373] INFO [c962b] {crash} Backtrace of thread 4 [SchedWorker]
2022-10-06T09:51:01Z [96373] INFO [308c3] {crash} frame  1 [+0x00000000016bd8c5] (anonymous namespace)::crashHandlerSignalHandler(int, siginfo_t*, void*) (+0x35)
2022-10-06T09:51:01Z [96373] INFO [308c3] {crash} frame  2 [+0x00000000039ac5a5] sigprocmask (+0x21)
2022-10-06T09:51:01Z [96373] INFO [308c3] {crash} frame  3 [+0x0000000000d909aa] arangodb::traverser::DepthFirstEnumerator::shouldPrune() (+0x3da)
2022-10-06T09:51:01Z [96373] INFO [308c3] {crash} frame  4 [+0x0000000000d91051] arangodb::traverser::DepthFirstEnumerator::next() (+0x221)
2022-10-06T09:51:01Z [96373] INFO [308c3] {crash} frame  5 [+0x0000000000ce9752] arangodb::aql::TraversalExecutor::doOutput(arangodb::aql::OutputAqlItemRow&) (+0x82)
2022-10-06T09:51:01Z [96373] INFO [308c3] {crash} frame  6 [+0x0000000000aec0ec] arangodb::aql::ExecutionBlockImpl<arangodb::aql::TraversalExecutor>::executeWithoutTrace(arangodb::aql::AqlCallStack const&) (+0x20c)
2022-10-06T09:51:01Z [96373] INFO [308c3] {crash} frame  7 [+0x0000000000aeef25] arangodb::aql::ExecutionBlockImpl<arangodb::aql::TraversalExecutor>::execute(arangodb::aql::AqlCallStack const&) (+0xb5)
2022-10-06T09:51:01Z [96373] INFO [308c3] {crash} frame  8 [+0x00000000019820de] arangodb::aql::SingleRowFetcher<(arangodb::aql::BlockPassthrough)1>::execute(arangodb::aql::AqlCallStack&) [clone .isra.0] (+0x90e)
2022-10-06T09:51:01Z [96373] INFO [308c3] {crash} frame  9 [+0x0000000000a6cd87] arangodb::aql::ExecutionBlockImpl<arangodb::aql::IdExecutor<arangodb::aql::SingleRowFetcher<(arangodb::aql::BlockPassthrough)1> > >::executeFetcher(arangodb::aql::ExecutionBlockImpl<arangodb::aql::IdExecutor<arangodb::aql::SingleRowFetcher<(arangodb::aql::BlockPassthrough)1> > >::ExecutionContext&, arangodb::aql::AqlCall const&) (+0x107)
2022-10-06T09:51:01Z [96373] INFO [308c3] {crash} frame 10 [+0x0000000000a6f4a0] arangodb::aql::ExecutionBlockImpl<arangodb::aql::IdExecutor<arangodb::aql::SingleRowFetcher<(arangodb::aql::BlockPassthrough)1> > >::executeWithoutTrace(arangodb::aql::AqlCallStack const&) (+0x1d70)
2022-10-06T09:51:01Z [96373] INFO [308c3] {crash} frame 11 [+0x0000000000a82915] arangodb::aql::ExecutionBlockImpl<arangodb::aql::IdExecutor<arangodb::aql::SingleRowFetcher<(arangodb::aql::BlockPassthrough)1> > >::execute(arangodb::aql::AqlCallStack const&) (+0xb5)
2022-10-06T09:51:01Z [96373] INFO [308c3] {crash} frame 12 [+0x00000000009601e6] arangodb::aql::Query::execute(arangodb::aql::QueryResult&) (+0x7a6)
2022-10-06T09:51:01Z [96373] INFO [308c3] {crash} frame 13 [+0x000000000066d069] arangodb::RestCursorHandler::registerQueryOrCursor(arangodb::velocypack::Slice const&) (+0x1429)
2022-10-06T09:51:01Z [96373] INFO [308c3] {crash} frame 14 [+0x000000000066db5c] arangodb::RestCursorHandler::execute() (+0x32c)
2022-10-06T09:51:01Z [96373] INFO [308c3] {crash} frame 15 [+0x00000000005c1496] arangodb::rest::RestHandler::executeEngine(bool) (+0xb6)
2022-10-06T09:51:01Z [96373] INFO [308c3] {crash} frame 16 [+0x00000000005c54fa] arangodb::rest::RestHandler::runHandlerStateMachine() (+0x3fa)
2022-10-06T09:51:01Z [96373] INFO [308c3] {crash} frame 17 [+0x00000000008cda9e] arangodb::Scheduler::WorkItem<arangodb::rest::CommTask::handleRequestSync(std::shared_ptr<arangodb::rest::RestHandler>)::{lambda()#1}>::invoke() (+0x12e)
2022-10-06T09:51:01Z [96373] INFO [308c3] {crash} frame 18 [+0x0000000000724b8f] arangodb::SupervisedScheduler::runWorker() (+0x2df)
2022-10-06T09:51:01Z [96373] INFO [308c3] {crash} frame 19 [+0x00000000016d15bd] arangodb::Thread::startThread(void*) (+0x6d)
2022-10-06T09:51:01Z [96373] INFO [308c3] {crash} frame 20 [+0x000000000172077a] ThreadStarter(void*) [clone .lto_priv.0] (+0xaa)
2022-10-06T09:51:01Z [96373] INFO [308c3] {crash} frame 21 [+0x00000000039b33dc] start (+0x70)
2022-10-06T09:51:01Z [96373] INFO [ded81] {crash} available physical memory: 33258086400, rss usage: 679845888, vsz usage: 1776168960, threads: 54
```

The strange thing is that the full (longer) version of this query executed from the Foxx service in fact works and returns correct result. But any attempt to execute it (the same or its reduced version, with very same parameters etc) from the _arangosh_ or web UI leads to a server crash.

**Comment by Simran-B** [comment]
User-defined functions and other functions that can only run on Coordinator nodes in a cluster are not supported in PRUNE expressions. For the upcoming 3.11, the following change has been made to reject queries with invalid PRUNE conditions:

> **Prevent moving of unsafe vertex or edge filters into the traversal**
Previously any vertex or edge filter condition qualified to be moved into the traversal. This is not safe if the condition is executed on DB servers. For example, on DB servers it is not possible to execute JavaScript user-defined functions, or some other AQL functions that only work on coordinators.

See #16622

**Comment by wajda** [comment]
@Simran-B can you explain why it works for me in 3.9? I'm doing FILTER+PRUNE with a user-defined function in the expression, on the cluster, and it seems to work... Does it mean the in my case the PRUNE expr isn't pushed to the db server and the whole traversal runs on a coordinator?

**Comment by Simran-B** [comment]
It's not expected to work in a cluster because PRUNE is designed to execute on DB-Servers, but UDFs can only run on Coordinators. However, it might work in a single server where everything is run on one machine. There are significant differences between 3.9, 3.10, and the upcoming 3.11 with regard to the traversal implementations. I'm not sure what might work in some but not other versions. Newer versions try to pull more filters into the traversal instead of post-filtering, which means more work is done on DB-Servers where UDFs cannot execute. When post-filtering, UDFs should be fine.

I always got a crash or assertion failure when I ran the second AQL query with the below arangosh test code, including 3.9 single server and cluster:
```js
require("@arangodb/aql/functions").register("t::x",d=>d.stop,true);
db._create("v", {numberOfShards:3});
db._createEdgeCollection("e", {numberOfShards:3});
db.v.save({_key:"A"});
db.v.save({_key:"B"});
db.v.save({_key:"C"});
db.e.save({_from:"v/A",_to:"v/B",stop:true});
db.e.save({_from:"v/B",_to:"v/C"});
db._query(`WITH v FOR v,e IN 1..10 OUTBOUND "v/A" e PRUNE e.stop RETURN v`);
db._query(`WITH v FOR v,e IN 1..10 OUTBOUND "v/A" e PRUNE T::X(e) RETURN v`);
```
```
2022-11-18T21:02:06Z [849] FATAL [a7902] {crash} 💥 ArangoDB 3.9.4 [linux], thread 5 [SchedWorker] caught unexpected signal 11 (SIGSEGV) accessing address 0x0000000000000008: signal handler invoked - image base address: 0x0000000000400000 - CPU context: rip: 0x0000000000f88600, rsp: 0x00007f697f9f4a20, efl: 0x0000000000010202, rbp: 0x00007f697f9f4b70, rsi: 0x00007f695897d548, rdi: 0x0000000000000003, rax: 0x0000000000000000, rbx: 0x00007f695681b480, rcx: 0x0000000000000030, rdx: 0x00007f697f9f4bcf, r8: 0x00007f6980f102d8, r9: 0x00007f695890e200, r10: 0x00007f697f9f4ba0, r11: 0x000000000000001b, r12: 0x00007f697f9f4c00, r13: 0x00007f695897d548, r14: 0x0000000000000000, r15: 0x00000000000003e8
2022-11-18T21:02:06Z [849] INFO [c962b] {crash} Backtrace of thread 5 [SchedWorker]
2022-11-18T21:02:06Z [849] INFO [308c3] {crash} frame  1 [+0x00000000016c73e5] (anonymous namespace)::crashHandlerSignalHandler(int, siginfo_t*, void*) (+0x35)
2022-11-18T21:02:06Z [849] INFO [308c3] {crash} frame  2 [+0x00000000039b55a5] sigprocmask (+0x21)
2022-11-18T21:02:06Z [849] INFO [308c3] {crash} frame  3 [+0x0000000000d969aa] arangodb::traverser::DepthFirstEnumerator::shouldPrune() (+0x3da)
2022-11-18T21:02:06Z [849] INFO [308c3] {crash} frame  4 [+0x0000000000d97051] arangodb::traverser::DepthFirstEnumerator::next() (+0x221)
2022-11-18T21:02:06Z [849] INFO [308c3] {crash} frame  5 [+0x0000000000cef5b2] arangodb::aql::TraversalExecutor::doOutput(arangodb::aql::OutputAqlItemRow&) (+0x82)
2022-11-18T21:02:06Z [849] INFO [308c3] {crash} frame  6 [+0x0000000000af1e2c] arangodb::aql::ExecutionBlockImpl<arangodb::aql::TraversalExecutor>::executeWithoutTrace(arangodb::aql::AqlCallStack const&) (+0x20c)
2022-11-18T21:02:06Z [849] INFO [308c3] {crash} frame  7 [+0x0000000000af4c65] arangodb::aql::ExecutionBlockImpl<arangodb::aql::TraversalExecutor>::execute(arangodb::aql::AqlCallStack const&) (+0xb5)
2022-11-18T21:02:06Z [849] INFO [308c3] {crash} frame  8 [+0x000000000198b9de] arangodb::aql::SingleRowFetcher<(arangodb::aql::BlockPassthrough)1>::execute(arangodb::aql::AqlCallStack&) [clone .isra.0] (+0x90e)
2022-11-18T21:02:06Z [849] INFO [308c3] {crash} frame  9 [+0x0000000000a72aa7] arangodb::aql::ExecutionBlockImpl<arangodb::aql::IdExecutor<arangodb::aql::SingleRowFetcher<(arangodb::aql::BlockPassthrough)1> > >::executeFetcher(arangodb::aql::ExecutionBlockImpl<arangodb::aql::IdExecutor<arangodb::aql::SingleRowFetcher<(arangodb::aql::BlockPassthrough)1> > >::ExecutionContext&, arangodb::aql::AqlCall const&) (+0x107)
2022-11-18T21:02:06Z [849] INFO [308c3] {crash} frame 10 [+0x0000000000a751c0] arangodb::aql::ExecutionBlockImpl<arangodb::aql::IdExecutor<arangodb::aql::SingleRowFetcher<(arangodb::aql::BlockPassthrough)1> > >::executeWithoutTrace(arangodb::aql::AqlCallStack const&) (+0x1d70)
2022-11-18T21:02:06Z [849] INFO [308c3] {crash} frame 11 [+0x0000000000a88635] arangodb::aql::ExecutionBlockImpl<arangodb::aql::IdExecutor<arangodb::aql::SingleRowFetcher<(arangodb::aql::BlockPassthrough)1> > >::execute(arangodb::aql::AqlCallStack const&) (+0xb5)
2022-11-18T21:02:06Z [849] INFO [308c3] {crash} frame 12 [+0x0000000000966156] arangodb::aql::Query::execute(arangodb::aql::QueryResult&) (+0x7a6)
2022-11-18T21:02:06Z [849] INFO [308c3] {crash} frame 13 [+0x00000000006728d9] arangodb::RestCursorHandler::registerQueryOrCursor(arangodb::velocypack::Slice const&) (+0x1429)
2022-11-18T21:02:06Z [849] INFO [308c3] {crash} frame 14 [+0x00000000006733cc] arangodb::RestCursorHandler::execute() (+0x32c)
2022-11-18T21:02:06Z [849] INFO [308c3] {crash} frame 15 [+0x00000000005c3176] arangodb::rest::RestHandler::executeEngine(bool) (+0xb6)
2022-11-18T21:02:06Z [849] INFO [308c3] {crash} frame 16 [+0x00000000005c71da] arangodb::rest::RestHandler::runHandlerStateMachine() (+0x3fa)
2022-11-18T21:02:06Z [849] INFO [308c3] {crash} frame 17 [+0x00000000008d66be] arangodb::Scheduler::WorkItem<arangodb::rest::CommTask::handleRequestSync(std::shared_ptr<arangodb::rest::RestHandler>)::{lambda()#1}>::invoke() (+0x12e)
2022-11-18T21:02:07Z [849] INFO [308c3] {crash} frame 18 [+0x00000000007087df] arangodb::SupervisedScheduler::runWorker() (+0x2df)
2022-11-18T21:02:07Z [849] INFO [308c3] {crash} frame 19 [+0x00000000016d99bd] arangodb::Thread::startThread(void*) (+0x6d)
2022-11-18T21:02:07Z [849] INFO [308c3] {crash} frame 20 [+0x000000000172a2fa] ThreadStarter(void*) [clone .lto_priv.0] (+0xaa)
2022-11-18T21:02:07Z [849] INFO [308c3] {crash} frame 21 [+0x00000000039bc3dc] start (+0x70)
2022-11-18T21:02:07Z [849] INFO [ded81] {crash} available physical memory: 16161554432, rss usage: 337580032, vsz usage: 1065893888, threads: 58
```

Does this also happen on your end with 3.9.4? What exactly does not crash the server for you?

**Comment by wajda** [comment]
> Does this also happen on your end with 3.9.4? What exactly does not crash the server for you?

No, nothing. I might have been mistaken, sorry. It indeed crashes on the test query as expected. Thank you for clarification.

I now also see the error message instead of a crash on the latest ArangoDB versions including 3.9.6, so the issue is solved. Thanks again!
